Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/ui/certificateselection/DccTicketingCertificateSelectionFragment.kt

```diff
@@ -1,24 +1,120 @@
+                            .actionDccTicketingCertificateSelectionFragmentToDccTicketingConsentTwoFragment(
+                                event.transactionContext,
+                                event.selectedCertificateContainerId
+                            )
+                    )
+                }
+            }
+        }
+
+        lifecycleScope.launchWhenStarted {
+            val certificatesList = viewModel.getCertificates()
+            certificatesAdapter.update(certificatesList)
+
+            binding.apply {
+                if (certificatesAdapter.itemCount == 0) {
+                    selectionTitle.text =
+                        getString(R.string.dcc_ticketing_certificate_selection_no_certificates_found_title)
+                    noCertificatesFound.visibility = View.VISIBLE
+                    allowedCertificates.visibility = View.INVISIBLE
+                    birthDate.visibility = View.INVISIBLE
+                    standardizedName.visibility = View.INVISIBLE
+                    requiredCertificatesTitle.text =
+                        getString(R.string.dcc_ticketing_certificate_selection_requested_certificates_text)
+                    faqLink.visibility = View.VISIBLE
+                    val link = when (Locale.getDefault().language) {
+                        Locale.GERMAN.language -> R.string.dcc_ticketing_faq_link_german
+                        else -> R.string.dcc_ticketing_faq_link_english
+                    }
+                    faqLink.setTextWithUrl(
+                        R.string.dcc_ticketing_certificate_selection_more_information_text,
+                        R.string.dcc_ticketing_faq_link_container,
+                        link
+                    )
+                } else {
+                    selectionTitle.text =
+                        getString(R.string.dcc_ticketing_certificate_selection_provider_requirements_title)
+                    noCertificatesFound.visibility = View.GONE
+                    allowedCertificates.text = args.transactionContext.accessTokenPayload?.vc?.type?.let {
+                        getRequestedCertificateTypes(
+                            requestedCertificatesList = it,
+                            context = requireContext(),
+                            separator = ","
+                        )
+                    } ?: ""
+                    birthDate.text = getString(R.string.dcc_ticketing_certificate_birthday)
+                        .format(args.transactionContext.accessTokenPayload?.vc?.dob)
+                    standardizedName.text = getFullName(
+                        args.transactionContext.accessTokenPayload?.vc?.fnt,
+                        args.transactionContext.accessTokenPayload?.vc?.gnt
+                    )
+                    requiredCertificatesTitle.text =
+                        getString(R.string.dcc_ticketing_certificates_meeting_requirements_title)
+                }
```

Comment:
can we improve it if constraint layout groups is used ?
noMatchingCertificatesGroup.isVisible = certificates.isEmpty 

matchingCertificatesGroup.isVisible = certificates.isNotEmpty 

I see many views set visibility based on that. and maybe small functions to make readable ?

Reply:
Added the groups in the xml's but haven't gotten the change to use them here.